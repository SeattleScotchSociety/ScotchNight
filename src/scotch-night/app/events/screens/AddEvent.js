// @flow
import React, { Component } from 'react';
import {
    Button,
    ScrollView,
    View,
    KeyboardAvoidingView,
    TextInput,
    StyleSheet,
    StatusBar
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as eventActions from '../EventActions';
import GrowingTextInput from '../../components/GrowingTextInput';

class AddEvent extends Component {
    static navigationOptions = {
        title: 'Add Event'
    };

    constructor(props) {
        super(props);

        this.state = {
            event: {
                id: '',
                date: '',
                location: '',
                description: ''
            }
        };

        this._handleOnAddPress = this._handleOnAddPress.bind(this);
    }

    _handleTextChange = (property, value) => {
        let { event } = this.state;

        event[property] = value;

        this.setState({ event });
    };

    _handleOnAddPress = () => {
        let { addEvent } = this.props.actions;
        const { navigate } = this.props.navigation;
        let { event } = this.state;

        addEvent(event);
        navigate('EventList', {});
    };

    render() {
        let { event } = this.state;

        return (
            <KeyboardAvoidingView
                behavior="padding"
                style={{
                    flex: 1,
                    justifyContent: 'center',
                    paddingHorizontal: 20,
                    paddingTop: 20
                }}>
                <ScrollView
                    style={{ flex: 1, backgroundColor: '#F8F8F9' }}
                    contentContainerStyle={{ paddingTop: 30 }}>

                    <View style={[styles.row, styles.firstRow]}>
                        <TextInput
                            value={event.date}
                            placeholder="Date"
                            onChangeText={this._handleTextChange.bind(this, 'date')}
                            style={styles.textInput}
                            autoCapitalize="words"
                            autoCorrect={false}
                            autoFocus={true}
                            blurOnSubmit={false}
                            returnKeyType="next"
                            onSubmitEditing={async () => {
                                this._locationInput.focus();
                            }}
                        />
                    </View>
                    <View style={styles.row}>
                        <TextInput
                            ref={view => {
                                this._locationInput = view;
                            }}
                            value={event.location}
                            placeholder="Location"
                            onChangeText={this._handleTextChange.bind(this, 'location')}
                            style={styles.textInput}
                            autoCapitalize="words"
                            autoCorrect={false}
                            blurOnSubmit={false}
                            returnKeyType="next"
                            onSubmitEditing={async () => {
                                this._descriptionInput.focus();
                            }}
                        />
                    </View>
                    <View style={styles.row}>
                        <GrowingTextInput
                            ref={view => {
                                this._descriptionInput = view;
                            }}
                            minHeight={80}
                            value={event.description}
                            placeholder="Description"
                            onChangeText={this._handleTextChange.bind(this, 'description')}
                            style={styles.growingTextInput}
                            autoCapitalize="sentences"
                            autoCorrect={false}
                            blurOnSubmit={false}
                        />
                    </View>
                    <Button
                        title="Save"
                        onPress={this._handleOnAddPress}
                        style={{
                            backgroundColor: 'maroon'
                        }}
                    />
                </ScrollView>

                <StatusBar barStyle="light-content" />
            </KeyboardAvoidingView>
        );
    }
}

function mapStateToProps(state) {
    return state;
}

function mapDispatchToProps(dispatch) {
    return { actions: bindActionCreators(eventActions, dispatch) };
}

export default connect(mapStateToProps, mapDispatchToProps)(AddEvent);

const styles = StyleSheet.create({
    headerText: {
        fontSize: 18,
        color: '#fff',
        fontWeight: '600'
    },
    row: {
        backgroundColor: '#fff',
        borderBottomWidth: StyleSheet.hairlineWidth,
        borderColor: '#ccc'
    },
    firstRow: {
        borderTopWidth: StyleSheet.hairlineWidth,
        borderColor: '#ccc'
    },
    textInput: {
        flex: 1,
        height: 45,
        paddingHorizontal: 15
    },
    growingTextInput: {
        paddingHorizontal: 15,
        paddingVertical: 15,
        fontSize: 15
    }
});
