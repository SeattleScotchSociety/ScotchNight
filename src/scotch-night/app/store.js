import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { AsyncStorage } from 'react-native';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer } from './Reducers';
import rootSaga from './Sagas';
import Bottles from './bottles';
let { loadBottles } = Bottles.Actions;

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
    reducer,
    undefined,
    compose(applyMiddleware(sagaMiddleware), autoRehydrate())
);

sagaMiddleware.run(rootSaga);

persistStore(store, { storage: AsyncStorage }, () => {
    store.dispatch(loadBottles());
}).purge();

export default store;
