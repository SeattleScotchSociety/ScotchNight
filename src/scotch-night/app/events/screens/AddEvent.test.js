import React from 'react';
import renderer from 'react-test-renderer';
import Store from '../../store';
import AddEvent from './AddEvent';

test('renders correctly', () => {
    const tree = renderer.create(<AddEvent store={Store} />).toJSON();
    expect(tree).toMatchSnapshot();
});
