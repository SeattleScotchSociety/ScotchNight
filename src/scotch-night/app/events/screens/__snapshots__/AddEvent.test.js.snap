// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingTop": 20,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#F8F8F9",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
          },
          Object {
            "borderColor": "#ccc",
            "borderTopWidth": 0.5,
          },
        ]
      }
    >
      <TextInput
        autoCapitalize="words"
        autoCorrect={false}
        autoFocus={true}
        blurOnSubmit={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Date"
        returnKeyType="next"
        style={
          Object {
            "flex": 1,
            "height": 45,
            "paddingHorizontal": 15,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
        }
      }
    >
      <TextInput
        autoCapitalize="words"
        autoCorrect={false}
        blurOnSubmit={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Location"
        returnKeyType="next"
        style={
          Object {
            "flex": 1,
            "height": 45,
            "paddingHorizontal": 15,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
        }
      }
    >
      <TextInput
        autoCapitalize="sentences"
        autoCorrect={false}
        blurOnSubmit={false}
        minHeight={80}
        multiline={true}
        onChange={[Function]}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        placeholder="Description"
        style={
          Array [
            Object {
              "fontSize": 15,
              "paddingHorizontal": 15,
              "paddingVertical": 15,
            },
            Object {
              "height": 80,
            },
          ]
        }
        underlineColorAndroid="rgba(0, 0, 0, 0)"
        value=""
      />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
